build-android/build_all.sh

```diff
@@ -45,9 +45,30 @@ findtool apksigner
 set -ev
 
 LAYER_BUILD_DIR=$PWD
-DEMO_BUILD_DIR=$PWD/../demos/android
 echo LAYER_BUILD_DIR="${LAYER_BUILD_DIR}"
-echo DEMO_BUILD_DIR="${DEMO_BUILD_DIR}"
+
+#
+# Android builds and c++ libraries:
+#
+#   https://developer.android.com/ndk/guides/cpp-support recommends using
+#   c++_shared for applications that use more than one shared library.
+#   If multiple libraries using c++_static are loaded several copies of
+#   the globals will be present in the C++ runtime. This also happens
+#   if the same library is dlopen/dlclosed several times, as when running
+#   the Layer Validation Tests. Some of the c++ runtime globals are
+#   thread_local, so each copy consumes a TLS key. There are only 128 TLS
+#   keys allowed on android, and the unit tests can hit this because of
+#   repeatedly loading and unloading VVL.
+#
+#   The drawback to using c++shared is that the layer library can no longer
+#   be installed manually, but must be installed in an APK. It is still
+#   common practice to load layer libraries manually, so this script will
+#   build the layer library using c++static by default.
+#
+# To build the layer libaries for running the layer validation tests on Android, c++_shared needs to be used for APP_STL. To do this set
+# C_LIB_TYPE to 'SHARED' on the following line:
+export C_LIB_TYPE=STATIC
```

Comment:
Allow the user to override C_LIB_TYPE

```suggestion
if [ -z "${C_LIB_TYPE}" ]; then
    export C_LIB_TYPE=STATIC
fi
```

Reply:
Great, done.